R/staggregate.R

```diff
@@ -228,11 +261,13 @@ staggregate_polynomial <- function(data, overlay_weights, daily_agg, time_agg =
   # Make each raster layer a data.table
   list_dt <- lapply(1:list_length, create_dt)
 
-  # Merge all data.tables together
+  # Merge all data.tables together if there are multiple
   clim_dt <- list_dt[[1]]
-  for(i in 2:list_length){
-    dt_m <- list_dt[[i]]
-    clim_dt <- merge(clim_dt, dt_m, by=c('x', 'y', 'date'))
+  if(list_length > 1 ){
+    for(i in 2:list_length){
+      dt_m <- list_dt[[i]]
+      clim_dt <- merge(clim_dt, dt_m, by=c('x', 'y', 'date'))
+    }
```

Comment:
I think we've toyed around with that idea. From what I can recall we ended up not doing it because the function calls lots of objects created within the larger function, and that adding them as arguments would be difficult because this function just takes x so we can use it in lapply.

Reply:
Oh right, yea that makes sense - let's just keep it as is then.